test/e2e/istio/probing_test.go

```diff
@@ -325,12 +326,7 @@ func setupGateway(t *testing.T, clients *test.Clients, names test.ResourceNames,
 		t.Fatalf("Failed to update Gateway %s/%s: %v", namespace, networking.KnativeIngressGateway, err)
 	}
 
-	var selectors []string
-	for k, v := range gw.Spec.Selector {
-		selectors = append(selectors, k+"="+v)
-	}
-	selector := strings.Join(selectors, ",")
-
+	selector := labels.SelectorFromSet(gw.Spec.Selector).String()
```

Comment:
This probably has more allocations and also it does a sort of the keys (while our keys might already be sorted) and we didn't need a sort before.

Reply:
This... is a test.

Reply:
test is code too 😉